addon/models/resource.js

```diff
@@ -388,12 +388,12 @@ const RelatedProxyUtil = Ember.Object.extend({
     @param {Ember.ObjectProxy|Ember.ArrayProxy} proxyFactory
     @return {PromiseProxy} proxy
   */
-  createProxy: function (resource, proxyFactory) {
+  createProxy: function (resource, type, proxyFactory) {
     const relation = this.get('relationship');
     const url = this.proxyUrl(resource, relation);
-    const service = resource.container.lookup('service:' + pluralize(relation));
+    const service = resource.container.lookup('service:' + pluralize(type));
     let promise = this.promiseFromCache(resource, relation, service);
-    promise = promise || service.findRelated(relation, url);
+    promise = promise || service.findRelated(type, url);
     let proxy = proxyFactory.extend(Ember.PromiseProxyMixin, {
       'promise': promise, 'type': relation
```

Comment:
From what I can tell, the type parameter on line 398 is not used at the moment, but I am probably missing something here...